Formula/a/appwrite.rb

```diff
@@ -0,0 +1,20 @@
+class Appwrite < Formula
+  require "language/node"
+
+  desc "Command-line tool for Appwrite"
+  homepage "https://appwrite.io"
+  url "https://github.com/appwrite/sdk-for-cli/archive/refs/tags/4.2.1.tar.gz"
+  sha256 "1ed066be23f9a8cd7d792221492ffa7d3122ccaf1b25642ebfee088e77ad1a5b"
+  license "BSD-3-Clause"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match "4.2.1", shell_output("#{bin}/appwrite --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Note that it's fine to also leave this version check in addition to adding a more substantial test. However, it will break on every upgrade since you've hardcoded the version. You probably want `assert_match version.to_s` instead.